src/components/EventTabs.tsx

```diff
@@ -0,0 +1,127 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import {
+    Content,
+    Text,
+} from '@adobe/react-spectrum';
+import { Item, Tabs } from '@react-spectrum/tabs';
+
+import EventList from './EventList';
+import { ZetkinEvent, ZetkinEventResponse } from '../types/zetkin';
+
+interface EventTabsProps {
+    bookedEvents: ZetkinEvent[] | undefined;
+    eventResponses?: ZetkinEventResponse[] | undefined;
+    onSignup?: (eventId: number, orgId: number) => void;
+    onUndoSignup: (eventId: number, orgId: number) => void;
+    timeRange: {
+        later: ZetkinEvent[] | undefined;
+        today: ZetkinEvent[] | undefined;
+        tomorrow: ZetkinEvent[] | undefined;
+        week: ZetkinEvent[] | undefined;
+    };
```

Comment:
Really happy you like the pull request!

I'll have a look and see if the bug can be fixed via the interface! Should I revert `getRespondEvents` to its previous state as well?

I agree `EventTabs` is quite verbose at the moment. It would be nice to be able to return `null` as part of ternaries instead, for instance.